### ISSUE TITLE
[navigator2 quickstart] Create a navigator2 quickstart

### ISSUE THREAD

**Issue by ericwindmill** [flutter/samples#2518]
Part of #2514

**Comment by justinmc** [comment]
I agree that we need a good example of this. I think it should include best practices like state restoration. I believe there are lots of Flutter apps out there that do not do state restoration properly, so you don't return to your previous page when switching back to the app after it was unloaded. I think people copy our trivial examples in the docs and build from there, missing these details.

**Comment by ericwindmill** [comment]
> I agree that we need a good example of this. I think it should include best practices like state restoration. I believe there are lots of Flutter apps out there that do not do state restoration properly, so you don't return to your previous page when switching back to the app after it was unloaded. I think people copy our trivial examples in the docs and build from there, missing these details.

I honestly haven't used nav2 much at all, so I may come to you with questions
